runtime/tests/checker/hashable_struct_test.go

```diff
@@ -0,0 +1,58 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package checker
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/onflow/cadence/runtime/sema"
+)
+
+func TestCheckHashableStruct(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+	   let a: HashableStruct = 1
+	   let b: HashableStruct = true
+	 `)
+
+	assert.NoError(t, err)
+}
+
+func TestCheckInvalidHashableStruct(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+	   resource R {}
+ 
+	   let a: HashableStruct = <-create R()
+	   let b: HashableStruct = [<-create R()]
+	   let c: HashableStruct = {1: true}
+	 `)
+
+	errs := RequireCheckerErrors(t, err, 3)
```

Comment:
Curious why there are three errors? shouldn't be only two? `let c: HashableStruct = {1: true}` should be OK, right?

Reply:
A Dictionary cannot be used as a key in a Dictionary right?

Reply:
It's just a `{Int: Bool}` dictionary. Or am I missing something?

Reply:
`Int` and `Bool` are `HashableStruct` types, but dictionaries (and other containers) are not (yet).

Reply:
Oh I see. Makes sense 👍

Reply:
Filed https://github.com/onflow/cadence/issues/2906 the idea.